src/JWT/ValidationParameters.cs

```diff
@@ -67,4 +69,13 @@ private ValidationParameters()
             TimeMargin = 0
         };
     }
+
+    public static class ValidationParametersExtensions
+    {
+        public static ValidationParameters With(this ValidationParameters @this, Action<ValidationParameters> action)

```

Comment:
I think you need to use a different name than `With<T>()`. Something like `SetValidationParameters` might work. I wonder if @abatishchev wants to more fully embrace the Fluent API style, adding a method per setting such as:

* `EnableValidateIssuerSigningKey`/`DisableValidateIssuerSigningKey`
* `EnableValidateLifetime`/`DisableValidateLifetime`
* etc.…

Could you specify a bit more clearly/fully what is acceptable, @abatishchev ? Thanks!

Reply:
I used `With` to make it synonymous with the `with` statements for `records` in `C# 9.0` (which, sadly, cannot be used here). Maybe it could be changed to something like `WithParameters`, but the `With` method seems clear enough

```csharp
var validationParameters = ValidationParameters.Default
    .With(parameters =>
    {
        parameters.ValidateLifetime = false;
        parameters.TimeMargin = 1;
    });
```

With regards to fully embracing the fluent api style, I think it would add additional maintanence without a lot of gain since the class is just a parameter class; any fluent methods would not have enough to do, they would just toggle the values of boolean parameters.